messaging/src/main/java/org/axonframework/common/ReflectionUtils.java

```diff
@@ -247,6 +247,18 @@ public static Method methodOf(Class<?> clazz, String methodName, Class<?>... par
         return primitiveWrapperType;
     }
 
+    /**
+     * Returns the boxed wrapper type for the given {@code type} if it is primitive.
+     *
+     * @param type a {@link Type} to return boxed wrapper type for
+     * @return the boxed wrapper type for the give {@code type}, or {@code type} if no wrapper class was found.
+     */
+    public static Type resolvePrimitiveWrapperTypeIfPrimitive(Type type) {
+        Assert.notNull(type, () -> "type may not be null");
+
+        return Optional.<Type>ofNullable(primitiveWrapperTypeMap.get(type)).orElse(type);
```

Comment:
Instead of using an Optional, you could do `primitiveWrapperTypeMap.getOrDefault(type, type)` instead.